### PR TITLE
Improve error handling/reporting in route.go

### PR DESCRIPTION
**Pull request by tpantelis**
- There were several call sites that did not check the error return value.
    Either propagate the error or at least log it.

  - The cleanXfrmPolicies and cleanRoutes functions return error but the callers do not
    check the error. It seems an error is not catastrophic to the callers so
    they do not need to handle it. Also each function has a loop in which an error could
    occur but the current behavior ignores it and continues. If we return the error then
    subsequent iterations of the loop would be elided. It seems it's safest to keep the
    current behavior but log the error. Also there doesn't seem to be any reason to
    return an error so that was removed

  - Added format args to some log messages to provide more context

  - Modified the wording in some log messages for consistency

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

### CODE REVIEW

**Comment by tpantelis** [comment]
No - I don't have the means yet to test it either.

File: pkg/routeagent/controllers/route/route.go [hunk 1]
```diff
@@ -256,14 +253,12 @@ func (r *RouteController) handleRemovedEndpoint(obj interface{}) {
 	if object, ok = obj.(*v1.Endpoint); !ok {
 		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
 		if !ok {
-			utilruntime.HandleError(fmt.Errorf("error decoding object, invalid type"))
-			klog.V(4).Infof("problem decoding object")
+			klog.Errorf("Error decoding object %v, invalid type", obj)
```
Comment (mpeterson) [inline pkg/routeagent/controllers/route/route.go]: To be in line with #28 

```suggestion
			klog.Errorf("Couldn't get object from tombstone %#v", obj)
```

File: pkg/routeagent/controllers/route/route.go [hunk 2]
```diff
@@ -256,14 +253,12 @@ func (r *RouteController) handleRemovedEndpoint(obj interface{}) {
...
 			return
 		}
 		object, ok = tombstone.Obj.(*v1.Endpoint)
 		if !ok {
-			utilruntime.HandleError(fmt.Errorf("error decoding object tombstone, invalid type"))
-			klog.V(4).Infof("problem decoding object tombstone")
+			klog.Errorf("Error decoding object tombstone %v, invalid type", tombstone.Obj)
```
Comment (mpeterson) [inline pkg/routeagent/controllers/route/route.go]: To be in line with #28 
```suggestion
			klog.Errorf("Could not convert object tombstone %v to an Endpoint", tombstone.Obj)
```

File: pkg/routeagent/controllers/route/route.go (hunk 1 above)
Comment (tpantelis) [reply pkg/routeagent/controllers/route/route.go]: yeah - we should be consistent. I'll change it.

Comment (mpeterson) [reply pkg/routeagent/controllers/route/route.go]: good, thanks!

**Pull request merged**